lib/MetaCPAN/Web/Controller/Author.pm

```diff
@@ -58,7 +58,17 @@ sub index : Chained('root') PathPart('') Args(0) {
             = $c->model('API::Favorite')->by_user( $author->{user} )->recv;
         $took += $faves_data->{took} || 0;
 
-        $faves = [ map { $_->{fields} } @{ $faves_data->{hits}->{hits} } ];
+        my @all_fav = map { $_->{fields}->{distribution} }
+            @{ $faves_data->{hits}->{hits} };
+        my $noLatest = $c->model('API::Release')->no_latest(@all_fav)->recv;
```

Comment:
I get why you did it -- it makes sense to me, but we don't really need to mimic the ES structure in this case (or worry about how long the query took).  The purpose of the model is to take the ES data structures and massage them into something we can easily work with.  

In this case the it's basically a question of whether or not a condition is true.  So for that, the structure above is a lot more complex than we need.

It's a big codebase with a lot of code to follow, so my inclination is to merge only what we need moving forward.  If we need more info at some point in the future, we can rework the model, but we're best off avoiding all increased complexity if we don't actually need it _right now_.  In this case returning an array of dist names would be my preference.  We can always use a map to create a hash if we need it.  Or we can return the hash, but the extra keys aren't adding value just yet.